### PR TITLE
Fix failing UI tests

### LINKED ISSUE #8341: Fix UI tests
UI tests are failing after they were merged into 'develop'. Tests need to be reviewed and fixed

### PR DESCRIPTION
**Pull request by brbrr**
Make sure tests are passing now. Also, fixed some warnings and done minor clean-up.

Fixes #8341 

To test:

Check buddybuild run

### CODE REVIEW

**Comment by brbrr** [comment]
Tests are passing in BuddyBuild: https://dashboard.buddybuild.com/apps/57a120bbe0f5520100e11c19/build/5a33d88f5074bb00017e4d03#tests

File: WordPress/WordPressUITests/LoginTests.swift [hunk 1]
```diff
@@ -55,6 +53,6 @@ class LoginTests: XCTestCase {
 //        let email = WordPressTestCredentials.nuxEmailPrefix + username + WordPressTestCredentials.nuxEmailSuffix
 //
 //        createAccount(email: email, username: username, password: WordPressTestCredentials.oneStepPassword)
-//        waitForElementToAppear(element: app.tabBars[ elementStringIDs.mainNavigationBar ], timeout: 20)
+//        waitForElementToAppear(element: XCUIApplication().tabBars[ elementStringIDs.mainNavigationBar ], timeout: 20)
```
Comment (SergioEstevao) [inline WordPress/WordPressUITests/LoginTests.swift]: Do we need this commented out code? Why keep it?

File: WordPress/WordPressUITests/Screens/BaseScreen.swift [hunk 2]
```diff
@@ -5,10 +5,10 @@ class BaseScreen {
     var app: XCUIApplication!
     var expectedElement: XCUIElement!
     var waitTimeout: Double!
-    var testCase: XCTestCase
+    public static var testCase: XCTestCase!
 
     init(element: XCUIElement) {
-        testCase = XCTestCase.init()
+//        testCase = XCTestCase.init()
```
Comment (SergioEstevao) [inline WordPress/WordPressUITests/Screens/BaseScreen.swift]: Why do we need to keep the commented out code.

File: WordPress/WordPressUITests/Screens/BaseScreen.swift [hunk 3]
```diff
@@ -23,13 +23,101 @@ class BaseScreen {
...
-        testCase.waitForExpectations(timeout: TimeInterval(timeout!), handler: nil)
+
+    func waitFor(element: XCUIElement, predicate: String, timeout: Int? = nil) {
+        let timeoutValue = timeout ?? 5
+
+        let elementPredicate = XCTNSPredicateExpectation(predicate: NSPredicate(format: predicate), object: element)
+        _ = XCTWaiter.wait(for: [elementPredicate], timeout: TimeInterval(timeoutValue))
+    }
+
+    func waitFor2(element: XCUIElement, predicate: String,
```
Comment (SergioEstevao) [inline WordPress/WordPressUITests/Screens/BaseScreen.swift]: Can we have a better name for this method? For example: ```waitAndLog(for:...```?

File: WordPress/WordPressUITests/Screens/BaseScreen.swift [hunk 4]
```diff
@@ -23,13 +23,101 @@ class BaseScreen {
...
+            }
+        }
     }
 
     func isLoaded() -> Bool {
         return expectedElement.exists
     }
+
+    class func waitForLoadingIndicatorToDisappear(within timeout: TimeInterval) {
+        #if os(tvOS)
```
Comment (SergioEstevao) [inline WordPress/WordPressUITests/Screens/BaseScreen.swift]: Do we need this? This project is not active for tvOS.

File: WordPress/WordPressUITests/Screens/LoginPasswordScreen.swift [hunk 5]
```diff
@@ -12,20 +12,29 @@ class LoginPasswordScreen: BaseScreen {
...
-        waitFor(predicate: "isEnabled == true", element: loginButton)
+        if loginButton.exists && !loginButton.isHittable {
+            waitFor(element: loginButton, predicate: "isEnabled == true")
+        }
         return self
     }
 
     func verifyLoginError() -> LoginPasswordScreen {
-        XCTAssert(XCUIApplication().staticTexts["pswdErrorLabel"].exists)
+        sleep(1) // Give some time for error text to appear
```
Comment (SergioEstevao) [inline WordPress/WordPressUITests/Screens/LoginPasswordScreen.swift]: Will not be better to wait for the label to appear with a timeout? Something like this: 

```
let app = XCUIApplication()
app.launch()

let label = app.staticTexts["Hello, world!"]
let exists = NSPredicate(format: "exists == 1")

expectationForPredicate(exists, evaluatedWithObject: label, handler: nil)
waitForExpectationsWithTimeout(5, handler: nil)
```

File: WordPress/WordPressUITests/LoginTests.swift (hunk 1 above)
Comment (brbrr) [reply WordPress/WordPressUITests/LoginTests.swift]: These tests are needed to be re-implemented. I've moved them into separate issue: #8356

File: WordPress/WordPressUITests/Screens/BaseScreen.swift (hunk 4 above)
Comment (brbrr) [reply WordPress/WordPressUITests/Screens/BaseScreen.swift]: It was just copy-paste from `WordPressScreenshotGeneration`. Removed ✅

**Pull request merged**